app/models/concerns/to_csv.rb

```diff
@@ -16,6 +16,18 @@ def to_csv
         dashboard.find_each(batch_size: 1_000) { |record| csv << record.dashboard_attributes.values }
       end
     end
+
+    # @param headers [Array<String>]
+    # @param data [Array<Array<String>>]
+    # @return [String]
+    def generate_csv(headers, data)
```

Comment:
Now we have this, can you see how `column_headers` and `dashboard` in the method above can be leveraged.

Reply:
OK. More like this? [b3688bd](https://github.com/DFE-Digital/early-years-foundation-recovery/pull/648/commits/b3688bd8919ab967a7cfbe9ba1aa9f0ffbf3d8a4)

Reply:
Nice one, but we can keep going.

- we can have a single point of entry for CSV generation
- the documented return values were incorrect `average_pass_scores` is a `Hash` and the `Array` included `Float`
- we can simplify the code by just calling `to_a` on that `Hash`

![image](https://github.com/DFE-Digital/early-years-foundation-recovery/assets/3261/c7203ad7-6aba-48b1-ac35-f8caf975298f)

We can achieve this by making `#to_csv` more flexible and able to handle Ruby primitives

![image](https://github.com/DFE-Digital/early-years-foundation-recovery/assets/3261/066d4e4d-b15f-4584-9f80-29c25289bac4)

I have these changes committed if you like I can push them.

Reply:
Yes please - thanks a lot @peterdavidhamilton

Reply:
Heads up. The next thing I'd like to discuss is consolidating the responsibility of coercing values and shared contexts for testing.

Reply:
Right, OK :+1: